File: questPresentation/src/main/java/ru/nekit/android/qls/quest/resources/struct/PairColorStruct.java
package ru.nekit.android.qls.quest.resources.struct;

import android.support.annotation.NonNull;
import android.util.Pair;

import ru.nekit.android.qls.quest.resources.collections.ColorResourceCollection;

public class PairColorStruct {

    private Pair<ColorResourceCollection, ColorResourceCollection> mData;

    public PairColorStruct(@NonNull ColorResourceCollection primaryColor,
                           @NonNull ColorResourceCollection secondaryColor) {
        mData = new Pair<>(primaryColor, secondaryColor);
    }

    public PairColorStruct(int primaryId,
                           int secondaryId) {
        mData = new Pair<>(ColorResourceCollection.getById(primaryId), ColorResourceCollection.getById(secondaryId));
    }

    @NonNull
    public ColorResourceCollection getPrimaryColorModel() {
        return mData.first;
    }

    @NonNull
    public ColorResourceCollection getSecondaryColorModel() {
        return mData.second;
    }
}

File: questPresentation/src/main/java/ru/nekit/android/qls/quest/resources/struct/PairColorStruct.kt
package ru.nekit.android.qls.quest.resources.struct

import android.util.Pair

import ru.nekit.android.qls.domain.model.resources.ColorResourceCollection

class PairColorStruct {

    private var data: Pair<ColorResourceCollection, ColorResourceCollection>? = null

    val primaryColorModel: ColorResourceCollection
        get() = data!!.first

    val secondaryColorModel: ColorResourceCollection
        get() = data!!.second

    constructor(primaryColor: ColorResourceCollection,
                secondaryColor: ColorResourceCollection) {
        data = Pair(primaryColor, secondaryColor)
    }

    constructor(primaryId: Int,
                secondaryId: Int) {
        data = Pair(ColorResourceCollection.getById(primaryId), ColorResourceCollection.getById(secondaryId))
    }
}

